[ui][docs] - `useNativeState` docs (#45282)
expo/expo: docs/pages/versions/unversioned/sdk/ui/jetpack-compose/usenativestate.mdx

---
title: useNativeState
description: A React hook that creates observable state shared between JavaScript and native Jetpack Compose views.
sourceCodeUrl: 'https://github.com/expo/expo/tree/main/packages/expo-ui/src/State/useNativeState.ts'
packageName: '@expo/ui'

> platforms: ['android']

Reviewer: ```suggestion
platforms: ['android', 'expo-go']
```

---

import APISection from '~/components/plugins/APISection';
import { APIInstallSection } from '~/components/plugins/InstallSection';

> `useNativeState` returns an `ObservableState` that maps to a Compose [`MutableState`](https://developer.android.com/reference/kotlin/androidx/compose/runtime/MutableState) on the native side, so reads and writes to `.value` are tracked directly by Compose without going through the React render cycle. This lets you update the native view synchronously from a worklet on the UI thread.

Reviewer: ```suggestion
`useNativeState` returns an [`ObservableState`](#observablestate) that maps to a Compose [`MutableState`](https://developer.android.com/reference/kotlin/androidx/compose/runtime/MutableState) on the native side, so reads and writes to `.value` are tracked directly by Compose without going through the React render cycle. This lets you update the native view synchronously from a worklet on the UI thread.
```

## Installation

<APIInstallSection />

## Usage

> > **Note:** Using worklets requires [`react-native-reanimated`](https://docs.swmansion.com/react-native-reanimated/) and [`react-native-worklets`](https://docs.swmansion.com/react-native-worklets/) to be installed in your project. `useNativeState` itself works without them, but the synchronous UI-thread updates shown below depend on the worklet runtime.

Reviewer 2: ```suggestion
> **Note:** Using worklets requires installing [`react-native-reanimated`](https://docs.swmansion.com/react-native-reanimated/) and [`react-native-worklets`](https://docs.swmansion.com/react-native-worklets/) in your project. `useNativeState` itself works without them, but the synchronous UI-thread updates shown below depend on the worklet runtime.
```
